Lib/test/test_codecs.py

```diff
@@ -531,6 +531,26 @@ def test_issue8941(self):
         self.assertEqual('\U00010000' * 1024,
                          codecs.utf_32_decode(encoded_be)[0])
 
+    def test_bug36819(self):
+        class State():
+            num = []
+
+            def __init__(self):
+                self.num.append(None)
+
+            def check(self):
+                return len(self.num)
+
+        def err(exc):
+            state = State()
+            # The interpreter should segfault after a handful of attempts.
+            # 50 was chosen to try to ensure a segfault without a fix,
+            # but not OOM a machine with one.
+            return ("a", exc.end if state.check() == 50 else exc.start)
+
+        codecs.register_error("err", err)
+        codecs.utf_32_encode("\udc80", "err")
```

Comment:
You can maybe add a test on len(State.num). I expect an exact number of calls.

Same remark for the utf_16_encode() test.